architecture/cicd_gitops.adoc

```diff
@@ -0,0 +1,62 @@
+
+[id="cicd_admin"]
+== CI/CD for cluster administration and application configuration management
+
+_Continuous integration_ is an automation process for developers. Code changes to an application are regularly built, tested, and merged to a shared repository.
+
+_Continuous delivery_ and _continuous deployment_ are closely related concepts that are sometimes used interchangeably that refer to automation of the pipeline.
+Continuous delivery uses automation to ensure that a developer’s changes to an application are tested and sent to a repository, where an operation team can deploy them to a production environment. Continuous deployment enables the release of changes, starting from the repository until production. Continuous deployment prevents the operations team from getting overloaded and speeds up application delivery.
+
+[id="cicd_gitops_methodology"]
+== The GitOps methodology and practice
+
+_GitOps_ is a set of practices that use Git pull requests to manage infrastructure and application configurations. The Git repository in GitOps is the only source of truth and contains the entire state of the system so that the trail of changes to the system state are visible and auditable. GitOps enables you to implement a DevOps methodology.
+
+You can use GitOps tooling to create repeatable and predictable processes for managing and recreating {product-title} clusters and applications. By using GitOps, you can address the issues of infrastructure and application configurations sprawl. It simplifies the propagation of infrastructure and application configurations changes across multiple clusters by defining your infrastructure and applications definitions as “code”. Implementing GitOps for your cluster configuration files can make automated installation easier and allow you to configure automated cluster customizations. You can apply the core principles of developing and maintaining software in a Git repo to the creation and management of your cluster and application configuration files.
+
+Automating both your cluster configuration and container development process by using GitOps allows you to pick and choose where and when to adopt this practice, using {product-title}. Using a CI pipeline that pairs with your GitOps strategy and execution plan is ideal. {product-title} provides the flexibility to choose when and how you integrate this methodology into your business practices and pipelines.
+
+GitOps works well with {product-title} because you can declaratively configure it and the state of the cluster configuration can be stored using Git. For more information, see xref:../installing/install_config/customizations.adoc#customizations[Available cluster customizations].
+
+[id="cicd_gitops_cluster_administration"]
+=== GitOps for single-cluster and multi-cluster administration
+
+Whether you need one or multiple independent or co-operative {product-title} clusters, you can use a GitOps strategy to manage the following tasks:
```

Comment:
s/Whether you need one or multiple independent or co-operative {product-title} clusters,/Whether you need one or more independent or cooperative clusters,/